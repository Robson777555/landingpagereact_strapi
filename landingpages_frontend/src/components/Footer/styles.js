import styled, { css } from 'styled-components';
import { Container as TextComponent } from '../../components/TextComponent/styles';
import { Container as SectionContainer } from '../SectionContainer/styles';

export const Container = styled.footer`
  /* eslint-disable-next-line no-unused-vars */
  ${({ theme }) => css`
    font-family:${theme.fonts.family.default};
    text-align: center;
    font-size: ${theme.fonts.sizes.small};
    border-top: 0.1rem solid ${theme.colors.mediumGray};
    
    a{
      color: inherit;
      text-decoration: none;
      
    }
    & ${TextComponent} {
      font-size: ${theme.fonts.sizes.small};
    }

    & ${SectionContainer} {
      padding-top: 0;
      padding-bottom: 0;
    }
  `}
  
`;
