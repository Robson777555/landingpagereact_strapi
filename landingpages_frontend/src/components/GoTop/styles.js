import styled, { css } from 'styled-components';

export const Container = styled.a`
  ${({ theme }) => css`
    position: fixed;
    background: ${theme.colors.primaryColor};
    color: ${theme.colors.white};
    display: flex;
    align-items: center;
    justify-content: center;
    width: 4rem;
    height: 4rem;
    bottom: 2rem;
    right: 2rem;
    border-radius: 50%;
    z-index: 6;
    opacity: 0.7;
    transition: all 0.3s ease;
    cursor: pointer;

    /* Melhorias para mobile */
    @media (max-width: 768px) {
      width: 3.5rem;
      height: 3.5rem;
      bottom: 1.5rem;
      right: 1.5rem;
    }

    &:hover {
      opacity: 1;
      transform: scale(1.05);
    }

    /* Garante que seja clic√°vel em mobile */
    -webkit-tap-highlight-color: transparent;
    touch-action: manipulation;
  `}
`;
