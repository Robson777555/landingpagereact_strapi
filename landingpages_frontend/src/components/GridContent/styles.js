import styled, { css } from 'styled-components';

export const Container = styled.div`
  /* eslint-disable-next-line no-unused-vars */
  ${({ theme }) => css`
    font-family:${theme.fonts.family.default}; 
    text-align: center;
    max-width: 58rem;
    margin: 0 auto;
    

  `}
`;

export const html = styled.div`
  /* eslint-disable-next-line no-unused-vars */
  ${({ theme }) => css`
    font-family:${theme.fonts.family.default}; 
    margin:${theme.spacings.xhuge} 0;
    text-align: center;
    
    pre {
      display: inline-block;
      
      margin: 0 auto;
    }

  `}
`;
