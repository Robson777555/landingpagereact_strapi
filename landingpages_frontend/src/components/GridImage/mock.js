export default {
  background: false,
  title: 'My grid 2',
  description: 'Tecnologias usadas na Landing Page:',
  grid: [
    {
      altText: 'Teste 1',
      srcImg:
        'https://storage.googleapis.com/medium-feed.appspot.com/images%2F9353691196%2F2f49650ed42b1-Para-que-e-usado-o-JavaScript.jpg?r=1',
    },
    {
      altText: 'Teste 2',
      srcImg:
        'https://devsagaz.com.br/content/images/2023/05/Ekran-Resmi-2019-11-18-18.08.13.png?r=2',
    },
    {
      altText: 'Teste 3',
      srcImg:
        'https://mms.businesswire.com/media/20200520005179/en/792790/23/Logo.WhiteBackground.jpg?r=3',
    },
    {
      altText: 'Teste 4',
      srcImg:
        'https://upload.wikimedia.org/wikipedia/commons/d/d9/Node.js_logo.svg?r=4',
    },
    {
      altText: 'Teste 5',
      srcImg:
        'https://blog.codapp.com.br/wp-content/uploads/2022/05/capa.jpg?r=5',
    },
    {
      altText: 'Teste 6',
      srcImg:
        'https://learn.microsoft.com/pt-br/shows/visual-studio-code/media/visual-studio-code-banner-image.jpg?r=6',
    },
  ],
};
