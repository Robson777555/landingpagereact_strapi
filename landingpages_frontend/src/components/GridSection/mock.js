export default {
  title: 'My grid',
  description: 'Esse site é para estudos de react e técnicas de programação',
  background: false,
  grid: [
    {
      title: 'Teste 1',
      description:
        'Pellentesque habitant morbi tristique senectus et netus et malesuada fames ac turpis egestas. Vestibulum tortor quam, feugiat vitae, ultricies eget, tempor sit amet, ante. Donec eu libero sit amet quam egestas semper. Aenean ultricies mi vitae est. Mauris placerat eleifend leo.',
    },
    {
      title: 'Teste 2',
      description:
        'Pellentesque habitant morbi tristique senectus et netus et malesuada fames ac turpis egestas. Vestibulum tortor quam, feugiat vitae, ultricies eget, tempor sit amet, ante. Donec eu libero sit amet quam egestas semper. Aenean ultricies mi vitae est. Mauris placerat eleifend leo.',
    },
    {
      title: 'Teste 3',
      description:
        'Pellentesque habitant morbi tristique senectus et netus et malesuada fames ac turpis egestas. Vestibulum tortor quam, feugiat vitae, ultricies eget, tempor sit amet, ante. Donec eu libero sit amet quam egestas semper. Aenean ultricies mi vitae est. Mauris placerat eleifend leo.',
    },
  ],
};
