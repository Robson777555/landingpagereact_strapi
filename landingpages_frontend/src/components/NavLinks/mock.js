export default [
  {
    children: 'My grid',
    link: '#target1',
    newTab: false,
  },

  {
    children: 'My grid 2',
    link: '#target2',
    newTab: false,
  },

  {
    children: 'Grid Two Collumns',
    link: '#target3',
    newTab: false,
  },
];
