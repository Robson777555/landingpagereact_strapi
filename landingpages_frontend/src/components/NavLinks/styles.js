import styled, { css } from 'styled-components';

export const Container = styled.nav`
  ${({ theme }) => css`
    display: flex;
    flex-direction: column;
    gap: ${theme.spacings.small};
    width: 100%;

    @media (min-width: 769px) {
      flex-direction: row;
      gap: ${theme.spacings.medium};
      width: auto;
    }

    a {
      color: ${theme.colors.primaryColor};
      text-decoration: none;
      font-size: ${theme.fonts.sizes.medium};
      padding: ${theme.spacings.small} 0;
      transition: color 0.3s ease;
      font-family: ${theme.fonts.family.default};

      &:hover {
        color: ${theme.colors.secondaryColor};
      }

      @media ${theme.media.lteMedium} {
        border-bottom: 1px solid ${theme.colors.mediumGray};
        text-align: center;
        padding: ${theme.spacings.small} ${theme.spacings.medium};
      }
    }
  `}
`;
