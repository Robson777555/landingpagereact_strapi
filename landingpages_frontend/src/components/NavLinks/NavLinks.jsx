import P from 'prop-types';
import { MenuLink } from '../MenuLink/MenuLink';
import * as Styled from './styles';

export const NavLinks = ({ links = [], onLinkClick }) => {
  return (
    <Styled.Container aria-label="Main menu">
      {links.map((link) => (
        <MenuLink key={link.link} {...link} onClick={onLinkClick} />
      ))}
    </Styled.Container>
  );
};

NavLinks.propTypes = {
  links: P.arrayOf(
    P.shape({
      children: P.string.isRequired,
      link: P.string.isRequired,
      newTab: P.bool,
    }),
  ),
  onLinkClick: P.func,
};
