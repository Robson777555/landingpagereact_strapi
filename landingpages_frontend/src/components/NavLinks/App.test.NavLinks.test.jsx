import { screen } from '@testing-library/react';
import { NavLinks } from './NavLinks';
import { renderTheme } from '../../styles/render-theme';
import { describe, it, expect } from '@jest/globals';
import mock from './mock';

describe('<NavLinks />', () => {
  it('should render links', () => {
    renderTheme(<NavLinks links={mock} />);
    expect(screen.getAllByRole('link')).toHaveLength(mock.length);
  });

  it('should not render links', () => {
    renderTheme(<NavLinks links={[]} />);
    expect(screen.queryAllByRole('link')).toHaveLength(0); // 🛠️ Corrigido
  });

  it('should render links media queries', () => {
    const { container } = renderTheme(<NavLinks links={mock} />);
    expect(container.firstChild).toHaveStyle({
      display: 'flex',
      'flex-flow': 'row wrap',
    });
  });

  it('should not render any links when links array is empty', () => {
    renderTheme(<NavLinks links={[]} />);
    expect(screen.queryAllByRole('link')).toHaveLength(0);
  });

  it('should render with default links prop', () => {
    renderTheme(<NavLinks />);
    expect(screen.queryAllByRole('link')).toHaveLength(0);
  });

  it('should match snapshot', () => {
    const { container } = renderTheme(<NavLinks links={mock} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
