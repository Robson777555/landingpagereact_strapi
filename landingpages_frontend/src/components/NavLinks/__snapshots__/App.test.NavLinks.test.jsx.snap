// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavLinks /> should match snapshot 1`] = `
.c1 {
  display: block;
  text-decoration: none;
  font-size: 1.6rem;
  padding: 0.8rem;
  color: #0A1128;
  position: relative;
  text-transform: uppercase;
}

.c1::after {
  content: '';
  position: absolute;
  bottom: 0.5rem;
  left: 50%;
  width: 0;
  height: 0.2rem;
  background-color: #dc143c;
  transition: all 300ms ease-in-out;
}

.c1:hover::after {
  left: 25%;
  width: 50%;
}

.c0 {
  display: flex;
  flex-flow: row wrap;
  align-items: center;
  justify-content: center;
  gap: 1rem;
}

@media (max-width: 768px) {
  .c1 {
    font-size: 2.4rem;
    padding: 2.4rem;
    text-align: center;
    min-height: 44px;
    display: block;
    width: 100%;
  }

  .c1::after {
    bottom: 0.5rem;
    width: 0;
  }

  .c1:hover::after {
    left: 25%;
    width: 50%;
  }
}

@media (max-width: 768px) {
  .c0 {
    flex-flow: column nowrap;
    align-items: stretch;
    width: 100%;
    margin-top: 2.4rem;
    max-height: 80vh;
    overflow-y: auto;
    padding-bottom: 3.2rem;
    gap: 0;
  }
}

<nav
  aria-label="Main menu"
  class="c0"
>
  <a
    class="c1"
    href="#target1"
    rel=""
    target="_self"
  >
    link 1
  </a>
  <a
    class="c1"
    href="#target2"
    rel=""
    target="_self"
  >
    link 2
  </a>
  <a
    class="c1"
    href="#target3"
    rel=""
    target="_self"
  >
    link 3
  </a>
  <a
    class="c1"
    href="#target4"
    rel=""
    target="_self"
  >
    link 4
  </a>
  <a
    class="c1"
    href="#target5"
    rel=""
    target="_self"
  >
    link 5
  </a>
  <a
    class="c1"
    href="#target6"
    rel=""
    target="_self"
  >
    link 6
  </a>
  <a
    class="c1"
    href="#target7"
    rel=""
    target="_self"
  >
    link 7
  </a>
  <a
    class="c1"
    href="#target8"
    rel=""
    target="_self"
  >
    link 8
  </a>
  <a
    class="c1"
    href="#target9"
    rel=""
    target="_self"
  >
    link 9
  </a>
  <a
    class="c1"
    href="#target10"
    rel=""
    target="_self"
  >
    link 10
  </a>
</nav>
`;
